site/profile/manifests/core/icinga_resources.pp

```diff
@@ -0,0 +1,784 @@
+# @summary
+#   Define and create all file resources of icinga master
+
+class profile::core::icinga_resources (
+  String $credentials_hash,
+  String $host_template,
+  String $http_template,
+  String $dns_template,
+  String $master_template,
+  String $ipa_template,
+){
+
+#<----------Variables Definition------------>
+#Implicit usage of facts
+  $master_fqdn  = $facts[fqdn]
+  $master_ip  = $facts[ipaddress]
+
+#Commands abreviation
+  $url_host    = "https://${master_fqdn}/director/host"
+  $url_svc     = "https://${master_fqdn}/director/service"
+  $credentials = "Authorization:Basic ${credentials_hash}"
+  $format      = 'Accept: application/json'
+  $curl        = 'curl -s -k -H'
+  $icinga_path = '/opt/icinga'
+  $lt          = '| grep Failed'
+
+#Service Templates Names
+  $http_svc_template_name    = 'HttpServiceTemplate'
+  $ping_svc_template_name    = 'PingServiceTemplate'
+  $dns_svc_template_name     = 'DnsServiceTemplate'
+  $master_svc_template_name  = 'MasterServiceTemplate'
+  $ipa_svc_template_name     = 'IpaServiceTemplate'
+  $disk_svc_template_name    = 'DiskServiceTemplace'
+
+#Service Names
+  $host_svc_ping_name   = 'HostPingService'
+  $host_svc_disk_name   = 'HostDiskService'
+  $dns_svc_name         = 'DnsService'
+  $dns_svc_ping_name    = 'DnsPingService'
+  $dns_svc_disk_name    = 'DnsDiskService'
+  $master_svc_dhcp_name = 'MasterDhcpService'
+  $master_svc_ping_name = 'MasterPingService'
+  $master_svc_disk_name = 'MasterDiskService'
+  $http_svc_name        = 'HttpService'
+  $http_svc_ping_name   = 'HttpPingService'
+  $http_svc_disk_name   = 'HttpDiskService'
+  $ipa_svc_name         = 'IpaService'
+  $ipa_svc_ping_name    = 'IpaPingService'
+  $ipa_svc_disk_name    = 'IpaDiskService'
+#<--------End Variables Definition---------->
+#
+#
+#<---------------JSON Files ---------------->
+##Hosts Templates JSON
+  $general_template_content = "{
```

Comment:
Use heredocs or load template files to do this.

Reply:
done as suggested